pages/draft/2019-09/migration-notes.md

```diff
@@ -0,0 +1,176 @@
+| `id`                     | `$id`                    | Core          | Identifier       | Already introduced in Draft 6, still used in Draft 2019-09.                          |
+| `examples`               | `examples`               | Annotation    | Informational    | Remains unchanged.                                                                   |
+| `readOnly` / `writeOnly` | `readOnly` / `writeOnly` | Annotation    | Informational    | Remains unchanged.                                                                   |
+
+
+
+### Migration Tutorial
+
+#### Step 1: Update `$schema` to use the new meta-schema
+
+Start by updating your `$schema` declaration to reference the correct Draft 2019-09 meta-schema:
+
+**Before (Draft 7):**
+
+```json
+{
+  "$schema": "http://json-schema.org/draft-07/schema#"
+}
+```
+
+**After (Draft 2019-09):**
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema"
+}
+```
+
+
+
+#### Step 2: Add `$vocabulary` if extending or customizing the schema
+
+Draft 2019-09 introduces a **vocabulary system** for declaring which keywords are in use. This is mostly relevant if you write custom meta-schemas or validators.
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2019-09/schema",
+  "$vocabulary": {
+    "https://json-schema.org/draft/2019-09/vocab/core": true,
+    "https://json-schema.org/draft/2019-09/vocab/applicator": true,
+    "https://json-schema.org/draft/2019-09/vocab/validation": true,
+    "https://json-schema.org/draft/2019-09/vocab/meta-data": true
+  }
+}
+```
+
+This helps tooling understand what features are supported in your schema. For basic schemas, it can be omitted unless you're writing meta-schemas.
+
+
+
+#### Step 3: Use `$recursiveAnchor` and `$recursiveRef` for recursive structures
+
+One of the biggest new features is **recursive referencing**, solving complex referencing for tree-like or self-nested structures.
+
+**Before (Draft 7) — using `$ref` with named anchors:**
+
+```json
+{
+  "$id": "node.schema.json",
+  "definitions": {
+    "node": {
+      "type": "object",
+      "properties": {
+        "value": { "type": "string" },
+        "children": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/node" }
+        }
+      }
+    }
+  }
+}
+```
+
+**After (Draft 2019-09) — using `$recursiveRef`:**
+
+```json
+{
+  "$id": "node.schema.json",
+  "$recursiveAnchor": true,
+  "type": "object",
+  "properties": {
+    "value": { "type": "string" },
+    "children": {
+      "type": "array",
+      "items": { "$recursiveRef": "#" }
+    }
+  }
+}
+```
+
+This allows cleaner recursive references and helps with schema composition across your documents.
+
+
+
+#### Step 4: Continue using supported keywords like `readOnly`, `writeOnly`, `examples`
+
+These keywords remain valid and unchanged:
```

Comment:
With the exception that they now emit annotations, which was not the case before